File: app/src/main/java/kotleni/b0mb3r/services/Service.java
package kotleni.b0mb3r.services;

import java.util.Random;

import okhttp3.Callback;
import okhttp3.OkHttpClient;

public abstract class Service {
    public static OkHttpClient client;

    public String phoneCode;
    public String phone;

    public String requireCode;

    private static String randomString(char min, char max, int length) {
        StringBuilder result = new StringBuilder();
        Random random = new Random();

        for (int i = 0; i < length; i++)
            result.append((char) (random.nextInt(max - min + 1) + min));

        return result.toString();
    }

    public static String getRussianName() {
        return Service.randomString('а', 'я', 5);
    }

    public static String getUserName() {
        return Service.randomString('a', 'z', 12);
    }

    public static String getEmail() {
        Random random = new Random();
        return getUserName() + "@" + new String[]{"gmail.com", "mail.ru", "yandex.ru"}[random.nextInt(3)];
    }

    public static String format(String phone, String mask) {
        StringBuilder formattedPhone = new StringBuilder();

        int index = 0;
        for (char symbol : mask.toCharArray()) {
            if (symbol == '*') {
                formattedPhone.append(phone.charAt(index));
                index++;
            } else {
                formattedPhone.append(symbol);
            }
        }

        return formattedPhone.toString();
    }

    public void setPhoneCode(String phoneCode) {
        this.requireCode = phoneCode;
    }

    public void prepare(String phoneCode, String phone) {
        this.phoneCode = phoneCode;
        this.phone = phone;
    }

    public String getFormattedPhone() {
        return phoneCode + phone;
    }

    public abstract void run(Callback callback);
}

File: app/src/main/java/kotleni/b0mb3r/services/Service.kt
package kotleni.b0mb3r.services

import okhttp3.OkHttpClient
import kotlin.random.Random


abstract class Service(var countryCodes: IntArray) {
    constructor(countryCode: Int) : this(intArrayOf(countryCode))
    constructor() : this(intArrayOf())

    abstract fun run(client: OkHttpClient, callback: Callback, phone: Phone)

    private fun randomString(min: Char, max: Char, length: Int): String {
        val allowedChars = (min..max)
        return (1..length)
            .map { allowedChars.random() }
            .joinToString("")
    }

    protected fun getRussianName(): String = randomString('9170562974', 'я', 5)
    protected fun getUserName(): String = randomString('a', 'z', 12)
    protected fun getEmail(): String = getUserName() + "@" + arrayOf("gmail.com", "mail.ru", "icloud.com")[Random.nextInt(3)]

    protected fun format(phone: String, mask: String): String {
        val formattedPhone = StringBuilder()
        var index = 0
        for (symbol in mask.toCharArray()) if (index < phone.length) formattedPhone.append(if (symbol == '*') phone[index++] else symbol)
        return formattedPhone.toString()
    }
}
